Nodejs/Product/Npm/SemverVersion.cs

```diff
@@ -60,12 +60,26 @@ public struct SemverVersion {
                 }
                 // /Hack
 
+                string preReleaseFragment;
+                if (!preRelease.Success || !IsValidOptionalFragment(preRelease.Value)) {
+                    preReleaseFragment = null;
+                } else {
+                    preReleaseFragment = preRelease.Value;
+                }
+
+                string buildMetadataFragment;
+                if (!buildMetadata.Success || !IsValidOptionalFragment(buildMetadata.Value)) {
+                    buildMetadataFragment = null;
+                } else {
+                    buildMetadataFragment = buildMetadata.Value;
+                }
+

```

Comment:
I don't see what value adding this code gets. The only difference over the original ternary operators is that is also calls IsValidOptionalFragment, but that just checks the fragment matches a RegEx which is identical to the match in the original RegEx that created the "prerelease" capture group. Thus is must be valid, else it wouldn't have been in the capture group and prerelease.Success would be false.